ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/sqlstd/SQLAuthorizationUtils.java

```diff
@@ -211,8 +211,15 @@ static RequiredPrivileges getPrivilegesFromMetaStore(IMetaStoreClient metastoreC
     RequiredPrivileges privs = getRequiredPrivsFromThrift(thrifPrivs);
 
     // add owner privilege if user is owner of the object
-    if (isOwner(metastoreClient, userName, curRoles, hivePrivObject)) {
-      privs.addPrivilege(SQLPrivTypeGrant.OWNER_PRIV);
+    try {
+      if (metastoreClient.tableExists(hivePrivObject.getDbname(), hivePrivObject.getObjectName()) &&
```

Comment:
PrivilageObjectType here might not be a TABLE_OR_VIEW.
````
    try {
      if (HivePrivilegeObjectType.TABLE_OR_VIEW != hivePrivObject.getType() ||
        metastoreClient.tableExists(hivePrivObject.getDbname(), hivePrivObject.getObjectName())) {

        if (isOwner(metastoreClient, userName, curRoles, hivePrivObject)) {
          privs.addPrivilege(SQLPrivTypeGrant.OWNER_PRIV);
        }
      }
    } catch (TException e) {
      throwGetPrivErr(e, hivePrivObject, userName);
    }
````

Reply:
Wait, don't you need to add the privilege when the table DOESN'T exist? Also won't it hide the problem with the non-existing table for other operations besides RENAME??

Reply:
- didn't you return 'true' in isOwner() when table doesn't exist in your initial patch and set the owner privilege?

Reply:
ack